From 34d96eff41c4c385378f4853d4235b80b989a9b5 Mon Sep 17 00:00:00 2001
From: Murala S Swamy <ssmurala@habana.ai>
Date: Fri, 23 Aug 2024 15:17:24 +0300
Subject: [PATCH] [FSW-8694] fs1: global fake fuse emulation flag

With this patch, added a global boot rom flag
which is used to identify fake/spi fuse emulation
already done. Upon any reset scenario boot rom
retrieve its value from cold flops. If flag
already set means boot rom can skip enumeration
again.

Change-Id: I67950fe410442b94e1de2562edfeec80baa87c6f
---

diff --git a/arch/fs1/arc/common.c b/arch/fs1/arc/common.c
index 23ffef6..a375802 100644
--- a/arch/fs1/arc/common.c
+++ b/arch/fs1/arc/common.c
@@ -142,6 +142,7 @@
 	boot_rom_global.efuse_read = cold_rst_data.efuse_read_flag;
 	boot_rom_global.boot_info_set_done = cold_rst_data.boot_info_set_done;
 	boot_rom_global.tpm_extend_done = cold_rst_data.tpm_validation_flag;
+	boot_rom_global.fuse_emulation_init_done = cold_rst_data.fuse_emulation_init_done;
 
 	if (fw_upd_status == FLOPS_UPD_PENDING)
 		boot_rom_global.fw_upd_switch_manifest = 1;
diff --git a/include/boot_rom.h b/include/boot_rom.h
index fd288c0..3d82d68 100644
--- a/include/boot_rom.h
+++ b/include/boot_rom.h
@@ -97,6 +97,7 @@
 	uint8_t die_type;
 	uint8_t sku;
 	uint64_t ppboot_start_addr;
+	bool fuse_emulation_init_done;
 	bool fake_fuse_en;
 	uint32_t fake_fuse_spi_offset;
 	bool boot_info_set_done;
